modules/core/src/main/java/org/apache/ignite/Ignition.java

```diff
@@ -490,7 +498,17 @@ public static Ignite ignite() throws IgniteIllegalStateException {
      * @return List of all grids started so far.
      */
     public static List<Ignite> allGrids() {
-        return IgnitionEx.allGrids();
+        List<Ignite> res = IgnitionEx.allGrids();
+
+        if (res == null || res.isEmpty())
+            return res;
+
+        Ignite ignite = res.get(0);
```

Comment:
We should check every node (or none of them, see comment about `IgniteEx`), but not only the first node.